packages/js/product-editor/src/components/product-select/types.ts

```diff
@@ -6,8 +6,19 @@ import { Product } from '@woocommerce/data';
 export type ProductSelectProps = {
 	className?: string;
 	placeholder?: string;
+	label?: string;
+	help?: JSX.Element | string | null;
 	selected: Product | Product[] | null;
 	items: Product[];
-	filter( value?: string ): Promise< Product[] >;
+	instanceNumber?: number;
+	isLoading?: boolean;
+	onAddNew?: ( value: string ) => void;
+	filter( value?: string ): Promise< Product[] > | void;
 	onSelect?( product: Product ): void;
```

Comment:
Made sure the new props are optional, incase plugins where already using the `ProductSelect` component. Otherwise it should work the same as before.